xwayland/server.c

```diff
@@ -310,19 +312,30 @@ static bool server_start(struct wlr_xwayland_server *server) {
 	server->client_destroy.notify = handle_client_destroy;
 	wl_client_add_destroy_listener(server->client, &server->client_destroy);
 
+	int p[2];
+	if (pipe(p) == -1) {
+		wlr_log_errno(WLR_ERROR, "pipe failed");
+		server_finish_process(server);
+		return false;
+	}
+	fcntl(p[1], F_SETFD, FD_CLOEXEC);
+	fcntl(p[0], F_SETFD, FD_CLOEXEC);
```

Comment:
We have a `set_cloexec` helper for this.

Reply:
Does using the helper gain us anything here? We're not interested in keeping any other flags in tact, and avoiding the call to F_GETFD reduces the size of the race window.

Reply:
We shouldn't assume that `FD_CLOEXEC` is the only flag, and we shouldn't change default flags. Additionally, we should error out if setting `FD_CLOEXEC` fails, instead of leaking a FD (leaking FDs are hard to debug).